src/gan/utilities/losses.py

```diff
@@ -0,0 +1,46 @@
+#!usr/bin/env/python

```

Comment:
This file feels very similar to something already reviewed; are both needed?

Reply:
This is where we define the adversarial loss for the GAN system. I'm not sure what this resembles - potentially an earlier version somewhere?